src/Savegame/BattleUnit.h

```diff
@@ -155,6 +155,7 @@ class BattleUnit
 	std::vector<std::pair<Uint8, Uint8> > _recolor;
 	bool _capturable;
 	ScriptValues<BattleUnit> _scriptValues;
+	std::vector<std::string> _tagData;
```

Comment:
what is point in storing this in each battle unit? Each unit will have exactly same list, this could have sens if would be very often used, but then it could be in other class keep to reduced copies of same data.

Reply:
I could not find a way to access the tag names other than this. Could the be queried from ScriptValues? It looked like they were just raw integer data to me, not associated with the tag names other than by index/id. So how would I go about having access to the names of the tags?

Reply:
Two solution, one is pass it to `getName(Language *lang, bool debugAppendId)` other is cache it in `Soldier`.
Aside from that, I could think for some helper function that transfor enum to string name, because this is third time where its used in code directly

Reply:
I figured as much. But getName is used in lots of places, iirc, so it would be quite an invasive change. And caching the tag names in soldier what was I was trying to achieve with the above solution. So what approach should I use?

Reply:
Ah, I misunderstood. I cached them in Unit and you are proposing caching them in Solider instead making it only one instance. That makes sense. I will try that route.